doc/en/user/source/community/cog/cog.rst

```diff
@@ -0,0 +1,134 @@
+.. _cog_plugin:
+
+COG (Cloud Optimized GeoTIFF) Support
+=====================================
+
+`COG <https://github.com/cogeotiff/cog-spec/blob/master/spec.md>`_ (Cloud Optimized GeoTIFF) is a regular GeoTIFF file, aimed at being hosted on a HTTP file server, whose internal organization is friendly for consumption by clients issuing HTTP GET range requests. 
+The COG module allows to set configuration params to connect to a Cloud GeoTIFF as well as adding JARs to the classpath needed to support that.
+
+Installation
+------------
+
+As a community module, the package needs to be downloaded from the `nightly builds <https://build.geoserver.org/geoserver/>`_,
+picking the community folder of the corresponding GeoServer series (e.g. if working on the GeoServer main development branch nightly
+builds, pick the zip file form ``master/community-latest``).
+
+To install the module, unpack the zip file contents into GeoServer own ``WEB-INF/lib`` directory and restart GeoServer.
+
+COG GeoTIFF Configuration Panel
+-------------------------------
+
+When configuring a GeoTIFF store, a new checkbox is available: ``Cloud Optimized GeoTIFF (COG)``. Setting that will open a new section presenting the COG configuration parameters for this COG Store.
+
+.. figure:: images/cogparams.png
+
+   COG Connection params
+
+Checking the ``Cloud Optimized GeoTIFF (COG)`` checkbox will open 
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Option
+     - Description
+   * - :guilabel:`URL`
+     - (prefixed by ``cog://``) representing the connection URL to the COG Dataset.
+   * - :guilabel:`Range Reader Settings`
+     - Which type of Range Reader implementation. Values Currently supported are HTTP and S3, the latter using an S3 Client
+   * - :guilabel:`User Name / Access Key ID`
+     - Optional user name (HTTP) or Access Key ID (S3) in case the COG dataset requires basic authentication
+   * - :guilabel:`Password / Secret Access Key`
+     - Password (HTTP) or Secret Access Key (S3) for the previous credential
+
+COG ImageMosaic Configuration
+-----------------------------
+New configuration parameters are available to be specified in the imageMosaic indexer configuration, in order to properly configure a COG based ImageMosaic.
```

Comment:
Additional configuration parameters can be specified in the ImageMosaic indexer configuration, in order to properly configure a COG based ImageMosaic.